downstairs/src/main.rs

```diff
@@ -401,53 +444,58 @@ async fn do_work_loop(
  */
 fn _show_work(ds: &Arc<Downstairs>) {
     let work = ds.work.lock().unwrap();
-    let mut kvec: Vec<u64> = work.active.keys().cloned().collect::<Vec<u64>>();
-    println!("--------------------------------------");
-    if kvec.is_empty() {
-        println!("Crucible Downstairs work queue:  Empty");
-    } else {
-        println!("Crucible Downstairs work queue:");
-        kvec.sort_unstable();
-        for id in kvec.iter() {
-            let dsw = work.active.get(id).unwrap();
-            let dsw_type;
-            let dep_list;
-            match &dsw.work {
-                IOop::Read {
-                    dependencies,
-                    eid: _eid,
-                    offset: _offset,
-                    num_blocks: _num_blocks,
-                } => {
-                    dsw_type = "Read ".to_string();
-                    dep_list = dependencies.to_vec();
-                }
-                IOop::Write {
-                    dependencies,
-                    eid: _eid,
-                    offset: _offset,
-                    data: _data,
-                } => {
-                    dsw_type = "Write".to_string();
-                    dep_list = dependencies.to_vec();
-                }
-                IOop::Flush {
-                    dependencies,
-                    flush_number: _flush_number,
-                } => {
-                    dsw_type = "Flush".to_string();
-                    dep_list = dependencies.to_vec();
-                }
-            };
-            println!(
-                "DSW:[{:04}] {} {:?} deps:{:?}",
-                id, dsw_type, dsw.state, dep_list,
-            );
+    for (uuid, upstairs_work) in work.iter() {
```

Comment:
Work is differentiated per UUID, so the answer's yes, but with only one active Upstairs we could also say the answer's no :)

The differentiation is so that reused job_ids don't cause issues when a new Upstairs becomes active. We could instead have a single work queue and scrub it when a new Upstairs becomes active. I'll see if that works out.

Reply:
Yeah, I think at this time we need a single work queue for a region.  The switch over should involve attempting to flush all outstanding, but be able to handle the case where it was not flushed.  Any time the downstairs looses connection to the upstairs, it has to throw away whatever it has in progress anyway, as the upstairs is responsible for consistency between the three region copies.

You still have to share the single work queue with different connections, but only one should ever be able to add work to it.